apm-sniffer/optional-plugins/trace-ignore-plugin/src/main/java/org/apache/skywalking/apm/plugin/trace/ignore/matcher/FastPathMatcher.java

```diff
@@ -122,9 +138,11 @@ private boolean wildcardMatch(String pat, int p, String str, int s) {
     }
 
     private boolean multiWildcardMatch(String pat, int p, String str, int s) {
-        // End of pattern, just check the end of string is '/' quickly.
-        if (p >= pat.length() && s < str.length()) {
-            return str.charAt(str.length() - 1) != '/';
+        switch (safeCharAt(pat, p)) {
+            // End of pattern, just return true quickly.
+            case 0: return true;
```

Comment:
If we get EOS when we enter multi-wildcard match state, return true quickly, because `**` mean anything